site/src/pages/WorkspacePage/WorkspacePage.tsx

```diff
@@ -1,46 +1,57 @@
 import { makeStyles } from "@material-ui/core/styles"
+import { useQuery } from "@tanstack/react-query"
 import { useMachine } from "@xstate/react"
+import { getWorkspaceBuildLogs } from "api/api"
+import { Workspace } from "api/typesGenerated"
 import { AlertBanner } from "components/AlertBanner/AlertBanner"
 import { ChooseOne, Cond } from "components/Conditionals/ChooseOne"
 import { Loader } from "components/Loader/Loader"
-import { FC, useEffect } from "react"
+import { FC, useRef } from "react"
 import { useParams } from "react-router-dom"
-import { firstOrItem } from "util/array"
 import { quotaMachine } from "xServices/quotas/quotasXService"
 import { workspaceMachine } from "xServices/workspace/workspaceXService"
 import { WorkspaceReadyPage } from "./WorkspaceReadyPage"
 import { RequirePermission } from "components/RequirePermission/RequirePermission"
 
+const useFailedBuildLogs = (workspace: Workspace | undefined) => {
+  const now = useRef(new Date())
+  return useQuery({
+    queryKey: ["logs", workspace?.latest_build.id],
+    queryFn: () => {
+      if (!workspace) {
+        throw new Error(
+          `Build log query being called before workspace is defined`,
+        )
+      }
```

Comment:
Instead of throwing an error, should we just stick this condition in the `enabled` key?

Reply:
We do, it is just to make TS happy!